velox/connectors/hive/storage_adapters/abfs/CMakeLists.txt

```
@@ -0,0 +1,33 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# for generated headers
+
+if(VELOX_ENABLE_ABFS)
+  add_library(velox_abfs AbfsReadFile.cpp AbfsFileSystem.cpp)
+  target_link_libraries(
+    velox_abfs
+    PUBLIC velox_file
+           velox_core
+           velox_hive_connector
+           velox_dwio_common_exception
+           Azure::azure-storage-blobs
+           Folly::folly
+           glog::glog
+           fmt::fmt)
```

Just adding `PUBLIC` is the same as not using the keyword ;)
```suggestion
    PUBLIC velox_file
           Azure::azure-storage-blobs
           Folly::folly
    PRIVATE velox_core
           velox_hive_connector
           velox_dwio_common_exception
           fmt::fmt
           glog::glog)
```

---

> Just adding `PUBLIC` is the same as not using the keyword ;)

ha, I thought make all PUBLIC is fine :),  what's the issue here if I don't make let's say `velox_core` private ?

---

It will the get passed along the dependency graph unnecessarily. So any target that links to this target inherits all PUBLICly linked libraries as dependencies and so on...

---

(and this is proper style and improves readability/comprehension of dependencies without having to trawl through the actual sources)